### PR TITLE
Add more tests.

### PR DESCRIPTION
**Pull request by maljub01**
Add tests to cover the following cases:
- Empty fields.
- Floating point numbers, and special floating point values.
- Integers.
- Maps.
- Repeated Fields.
- Other types: nil, bool, bytes.

Also, add a `testCase` struct type which is used to simplify the tables
for test cases and share the test validation logic.

### CODE REVIEW

File: tests/testcase.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import (
+	"fmt"
+
+	"github.com/benlaurie/objecthash/go/objecthash"
+	"github.com/golang/protobuf/proto"
+)
+
+type testCase struct {
+	protos               []proto.Message
```
Comment (getagit) [inline tests/testcase.go]: Comment this field.

File: tests/repeated_fields.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+				&pb3_latest.Repetitive{FloatField: []float32{0}},
+			},
+			equivalentJsonString: "{\"float_field\": [0]}",
+			equivalentObject:     map[string][]float32{"float_field": []float32{0}},
+			expectedHashString:   "63b09f87ed057a88b38e2a69b6dde327d9e2624384542853327d6b90c83046f9",
+		},
+
+		{
+			protos: []proto.Message{
+				&pb2_latest.Repetitive{FloatField: []float32{-2, -1, 0, 1, 2}},
```
Comment (getagit) [inline tests/repeated_fields.go]: How about some actual floating point numbers too?

Comment (maljub01) [reply tests/repeated_fields.go]: Done.

File: tests/testcase.go (hunk 1 above)
Comment (maljub01) [reply tests/testcase.go]: Done.

File: tests/repeated_fields.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+		/////////////////////////
+		//  Lists with floats. //
+		/////////////////////////
+		{
+			protos: []proto.Message{
+				&pb2_latest.Repetitive{FloatField: []float32{0}},
+				&pb3_latest.Repetitive{FloatField: []float32{0}},
+			},
+			equivalentJsonString: "{\"float_field\": [0]}",
```
Comment (getagit) [inline tests/repeated_fields.go]: How about a 0.0 as well?

File: tests/empty_fields.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+				// Proto3 scalar fields set to their default values are considered empty.
+				&pb3_latest.Simple{BoolField: false},
+				&pb3_latest.Simple{BytesField: []byte{}},
+				&pb3_latest.Simple{DoubleField: 0},
+				&pb3_latest.Simple{Fixed32Field: 0},
+				&pb3_latest.Simple{Fixed64Field: 0},
+				&pb3_latest.Simple{FloatField: 0},
+				&pb3_latest.Simple{Int32Field: 0},
+				&pb3_latest.Simple{Int64Field: 0},
```
[the remark this answers is not in the document]
Comment (maljub01) [reply tests/empty_fields.go]: Done.

File: tests/repeated_fields.go (hunk 3 above)
Comment (maljub01) [reply tests/repeated_fields.go]: Done.

**Pull request merged**